docs/rackhd/ubuntu_package_installation.rst

```diff
@@ -0,0 +1,232 @@
+
+Set up the RackHD bintray repository for use within this instance of Ubuntu
+
+.. code::
+
+    echo "deb https://dl.bintray.com/rackhd/debian trusty main" | sudo tee -a /etc/apt/sources.list
+    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 379CE192D401AB61
+    sudo apt-get update
+
+Install the RackHD Packages. Note: these packages are rebuilt on every commit to master and are
+not explicitly versioned, but intended as a means to install or update to the latest code most
+conveniently.
+
+.. code::
+    sudo apt-get install on-dhcp-proxy on-http on-taskgraph
+    sudo apt-get install on-tftp on-syslog
+
+Update dhcpd.conf per your network configuration
+
+.. code::
+
+    # RackHD added lines
+    deny duplicates;
+
+    ignore-client-uids true;
+
+    subnet 172.31.128.0 netmask 255.255.252.0 {
+      range 172.31.128.2 172.31.131.254;
+      # Use this option to signal to the PXE client that we are doing proxy DHCP
+      option vendor-class-identifier "PXEClient";
+    }
+
+
+The services files in /etc/init/ all need a conf file to exist in /etc/default/{service}
+Touch those files to allow the upstart scripts to start automatically.
+
+.. code::
+
+    for service in $(echo "on-dhcp-proxy on-http on-tftp on-syslog on-taskgraph");
+    do touch /etc/default/$service;
+    done
+
+#######
+
+Created the required file /opt/onrack/etc/config.json , you can use the demonstration
+configuration file at https://github.com/RackHD/RackHD/blob/master/packer/ansible/roles/monorail/files/config.json
+as a reference.
+
+
+Downloaded binary files from bintray.com/rackhd/binary and placed them using https://github.com/RackHD/RackHD/blob/master/packer/ansible/roles/images/tasks/main.yml as a guide.
+
+.. code::
+
+    #!/bin/bash
+
+    mkdir -p /var/renasar/on-tftp/static/tftp
+    cd /var/renasar/on-tftp/static/tftp
+
+    for file in $(echo "\
+    undionly.kpxe \
+    monorail.ipxe \
+    monorail-efi64-snponly.efi \
+    monorail-efi32-snponly.efi");do
+    wget "https://dl.bintray.com/rackhd/binary/ipxe/$file"
+    done
+
+    mkdir -p /var/renasar/on-http/static/http/common
+    cd /var/renasar/on-http/static/http/common
+
+    for file in $(echo "\
+    base.trusty.3.13.0-32-generic.squashfs.img \
+    base.trusty.3.16.0-25-generic.squashfs.img \
+    discovery.overlay.cpio.gz \
+    initrd.img-3.13.0-32-generic \
+    initrd.img-3.16.0-25-generic \
+    vmlinuz-3.13.0-32-generic \
+    vmlinuz-3.16.0-25-generic");do
+    wget "https://dl.bintray.com/rackhd/binary/builds/$file"
+    done
+
+
+
+All the services are upstart and have logs in /var/log/upstart.  Start with 'start on-[something]'
+Verify with 'ps | aux | grep node'
+
+
+**Set the Links to the Mirrors:**
+
+  .. code::
+
+    sudo ln -s /var/mirrors/esxi <on-http directory>/static/http/vmware
+    sudo ln -s /var/mirrors/esxi <on-tftp directory>/static/tftp/vmware
+    sudo ln -s /var/mirrors/ubuntu_trusty <on-http directory>/static/http/ubuntu_trusty
```

Comment:
When we updated the Ubuntu Trusty ipxe and preseed, it was done based on the assumption that /ubuntu was being proxied (instead of ubuntu_trusty being mirrored).

I think even if we did mirror, I'd rather mirror as /ubuntu and create the full directory structure (`dists/trusty/main/installer-amd64/current/images/netboot/ubuntu-installer/amd64`). I wouldn't mirror all distros, just the ones I'm interested in. But I wouldn't make a `ubuntu_` directory for each version.  It's like with Centos, you have 6.5 and 7 under /centos.